conference.js

```diff
@@ -2197,18 +2215,27 @@ export default {
         APP.UI.addListener(
             UIEvents.AUDIO_DEVICE_CHANGED,
             micDeviceId => {
+                const audioWasMuted = this.isLocalAudioMuted();
+
                 sendEvent(
                     'settings.changeDevice.audioIn');
                 createLocalTracksF({
                     devices: [ 'audio' ],
                     cameraDeviceId: null,
                     micDeviceId
                 })
-                .then(([ stream ]) => {
-                    this.useAudioStream(stream);
+                .then(([ stream ]) => this.useAudioStream(stream))
+                .then(() => {
                     logger.log('switched local audio device');
-                    APP.settings.setMicDeviceId(micDeviceId, true);
+
+                    // If audio was muted before changing current device
+                    // and selected new one, then mute new audio track.
+                    if (audioWasMuted) {
+                        sendEvent('changeDevice.audioIn.muted');
+                        muteLocalAudio(true);
```

Comment:
I noticed similar logic in replaceLocalTrack. I wonder if playing around with that logic produces a similar result to your changes.

Reply:
replaceLocalTrack is not taking care of it now. Maybe we can fix that ... I'm thinking that all this can be improved when we move all the conference.js logic into react, WDYT?

Reply:
muteLocalAudio/muteLocalVideo dispatch setVideoMuted/setAudioMuted. replaceLocalTrack also dispatches setVideoMuted/setAudioMuted. It seems like there is repeated logic going on and I'm still wondering if it's simple enough to remove. But it is very well likely that it looks simpler than it actually is.

Reply:
I'm looking into it now....

Reply:
Looks like, at least with video, you can mute the track before calling the action replaceLocalTrack, then within replaceLocalTrack dispatch setMuted() but pass in the current mute state of the track, and remove the second muting that has been added. I haven't tested it well but it seems to work on the happy path and I didn't see a flag of the remote participant's video while it was switching.